src/Text/Pandoc/Writers/XWiki.hs

```diff
@@ -0,0 +1,256 @@
+
+blockToXWiki (DefinitionList items) = do
+  lev <- asks listLevel
+  contents <- local (\s -> s { listLevel = listLevel s <> ";" }) $ mapM definitionListItemToMediaWiki items
+  return $ vcat contents <> if Text.null lev then "\n" else ""
+
+-- TODO: support more features
+blockToXWiki (Table _ _ _ headers rows') = do
+  headers' <- mapM (tableCellXWiki True) headers
+  otherRows <- mapM formRow rows'
+  return $ Data.Text.unlines (Data.Text.unwords headers':otherRows)
+
+formRow :: PandocMonad m => [[Block]] -> XWikiReader m Text
+formRow row = do
+  cellStrings <- mapM (tableCellXWiki False) row
+  return $ Data.Text.unwords cellStrings
+
+
+tableCellXWiki :: PandocMonad m => Bool -> [Block] -> XWikiReader m Text
+tableCellXWiki isHeader cell = do
+  contents <- blockListToXWiki cell
+  let cellBorder = if isHeader then "|=" else "|"
+  return $ cellBorder <> contents
+  
+
+inlineListToXWiki :: PandocMonad m => [Inline] -> XWikiReader m Text
+inlineListToXWiki lst =
+  fmap Data.Text.concat $ mapM inlineToXWiki lst
+
+
+inlineToXWiki :: PandocMonad m => Inline -> XWikiReader m Text
+
+inlineToXWiki (Str str) = return $ escapeXWikiString $ pack str
+
+inlineToXWiki Space = return " "
+
+inlineToXWiki LineBreak = return "\n"
+
+-- FIXME: better understand this
+inlineToXWiki SoftBreak = return " "
+
+inlineToXWiki (Emph lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "//" <> contents <> "//"
+
+inlineToXWiki (Strong lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "**" <> contents <> "**"
+
+inlineToXWiki (Strikeout lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "--" <> contents <> "--"
+
+inlineToXWiki (Superscript lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "^^" <> contents <> "^^"
+
+inlineToXWiki (Subscript lst) = do
+  contents <- inlineListToXWiki lst
+  return $ ",," <> contents <> ",,"
+
+-- TODO: Not supported. Maybe escape to HTML?
+inlineToXWiki (SmallCaps lst) = do
+  contents <- inlineListToXWiki lst
+  return contents
+
+inlineToXWiki (Quoted SingleQuote lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "'" <> contents <> "'"
+
+inlineToXWiki (Quoted DoubleQuote lst) = do
+  contents <- inlineListToXWiki lst
+  return $ "\"" <> contents <> "\""
+
+inlineToXWiki (Code (_,classes,_) contents') = do
+  let at  = Set.fromList classes `Set.intersection` highlightingLangs
+  let contents = pack contents'
+  return $
+    case Set.toList at of
+      [] -> "{{code}}" <> contents <> "{{/code}}"
+      (l:_) -> "{{code language=\"" <> (pack l) <> "\"}}" <> contents <> "{{/code}}"
+
+inlineToXWiki (Cite _ lst) = inlineListToXWiki lst
+
+-- FIXME: optionally support this (plugin?) 
+inlineToXWiki (Math _ str) = return $ "{{formula}}" <> (pack str) <> "{{/formula}}"
+
+inlineToXWiki il@(RawInline frmt str)
+  | frmt == Format "xwiki" = return $ pack str
+  | otherwise              = "" <$ report (InlineNotRendered il)
+
+-- TODO: Handle anchors
```

Comment:
It would be good to handle internal links.

Reply:
Can you give me an example of what a `Link` to an internal reference would look like? I don't see anything in the testsuite.native file.

Reply:
The following markdown should result in a sensible xwiki representation with `pandoc input.md -t xwiki`

```
# title {#myid}

some internal link that [links to the tile](#myid)
```

Reply:
Test produces reasonable output, but only if the user adds the "id" extension:

https://extensions.xwiki.org/xwiki/bin/view/Extension/Id%20Macro

There's no default support for arbitrary anchors in xwiki.